pkg/pilot/elasticsearch/v5/controller.go

```diff
@@ -1,11 +1,41 @@
 package v5
 
 import (
+	"context"
+	"fmt"
+
 	"github.com/jetstack/navigator/pkg/apis/navigator/v1alpha1"
 )
 
+const (
+	elasticsearchLocalNodeID = "_local"
+)
+
 func (p *Pilot) syncFunc(pilot *v1alpha1.Pilot) error {
-	if p.genericPilot.IsThisPilot(pilot) {
+	// TODO: perform cluster wide actions if we are a leader
+	if !p.genericPilot.IsThisPilot(pilot) {
+		return nil
+	}
+	if pilot.Status.Elasticsearch == nil {
+		pilot.Status.Elasticsearch = &v1alpha1.ElasticsearchPilotStatus{}
+	}
+	// set the ES document count to nil to indicate an unknown number of
+	// documents in case obtaining the document count fails. This prevents the
+	// node being shut down if it is not safe to do so.
+	pilot.Status.Elasticsearch.Documents = nil
+	if p.localESClient == nil {
+		return fmt.Errorf("local elasticsearch client not available")
+	}
```

Comment:
Thanks for explaining this, in https://github.com/jetstack/navigator/pull/131#pullrequestreview-77386340.
Did you see https://github.com/olivere/elastic/wiki/Retrier-and-Backoff? Maybe if you configure that, the `NewClient` won't immediately fail.
Also consider creating a new `NewSimpleClient` here each time: https://github.com/olivere/elastic/blob/release-branch.v6/client.go#L331

Reply:
`p.localESClient` is also called in `probes.go` . Does that need the same nil pointer check?

Reply:
Yep it will also need that check - thanks for pointing it out. I'd like to come up with a nicer way to do this :(

& thanks for the links - I'll have a read of those!